### ISSUE TITLE
rustdoc: #[doc(hidden)] attribute on trait method implemention causes it to be "unhidden"

### ISSUE THREAD

**Issue by QuietMisdreavus** [rust-lang/rust#55686]
While investigating https://github.com/rust-lang/rust/issues/36035, i noticed that that sample looked a bit odd:

![image](https://user-images.githubusercontent.com/5217170/47973662-3def1000-e06b-11e8-88b0-64fa64dd9820.png)

If you remove the `#[doc(hidden)]` from the implementation, the `fn hidden()` goes back under the "hidden undocumented items" section.

cc @GuillaumeGomez

**Comment by GuillaumeGomez** [comment]
You mean it should be in the "hidden undocumented items" section?

**Comment by QuietMisdreavus** [comment]
Yes, but the situation may be more complicated than it seems. I left a longer comment in the issue i linked in the OP (https://github.com/rust-lang/rust/issues/36035#issuecomment-435735090) that describes what i think is going on.

**Comment by steveklabnik** [comment]
Triage: any chance you've got the code that reproduces this issue?

**Comment by fmease** [comment]
The _Show hidden undocumented items_ subsection does not exist anymore in today's rustdoc, right?
So this issue is essentially fixed.

**Comment by GuillaumeGomez** [comment]
Indeed. Closing.
